### DISCUSSION TITLE
temporary parking segment creates double header

### DISCUSSION THREAD

**Discussion by maxxer** [nextgenhealthcare/connect#5159] [Q&A] [answered]
Hi. I made a JS transformer which 

```
var OBXseg = msg['OBX'].toString()
delete msg['OBX'];
```
does other stuff then
```
tmp['OBX'] = SerializerFactory.getSerializer('HL7V2').fromXML(OBXseg);
```

Unfortunately, the generated message has a double OBX segment this way. Source:
```
OBX||ED|TEST...
```
transformed:
```
OBX|OBX||ED|TEST...
```
How can I strip away the double OBX in the *new* segment?

Thanks

**Comment by jonbartels** [comment, marked as answer]
You shouldn't need to call toString nor serialize `OBXseg` . 

I think this would work.
```
var OBXseg = msg['OBX'];
..dooo stuffffff....
tmp['OBX'] =OBXseg;
delete msg['OBX'];
```

This solves your problem because you're getting the double-OBX because you're effectively calling `tmp['OBX'] = "<obx>....<obx>"`
